packages/overlays/src/OverlayMixin.js

```diff
@@ -15,52 +16,110 @@ export const OverlayMixin = dedupeMixin(
             type: Boolean,
             reflect: true,
           },
-          popperConfig: Object,
+          config: {
+            type: Object,
+          },
+          closeEventName: {
+            type: String,
+          },
         };
       }
 
+      constructor() {
+        super();
+        this.config = {};
+        this.closeEventName = 'overlay-close';
+      }
+
       get opened() {
         return this._overlayCtrl.isShown;
       }
 
       set opened(show) {
+        if (show) {
+          this.dispatchEvent(new Event('before-show'));
+        }
         this._opened = show; // mainly captured for sync on connectedCallback
         if (this._overlayCtrl) {
           this.__syncOpened();
         }
       }
 
-      __syncOpened() {
-        if (this._opened) {
-          this._overlayCtrl.show();
-        } else {
-          this._overlayCtrl.hide();
-        }
+      get config() {
+        return this._config;
       }
 
-      get popperConfig() {
-        return this._popperConfig;
+      set config(value) {
```

Comment:
having a setter for config means it is not retriggering a render automatically...

possible solutions
1) call `this.requestUpdate()` in setter
2) do not have manual setter/getter and move logic into `_requestUpdate`

Reply:
That is actually a conscious decision. The config is strictly for features of the overlay behavior and positioning. It should not trigger a re-render, because the config is completely decoupled from the template of the component. For example, invokerNode and configNode are not "meant" to be part of this `config` property. Only features / viewportConfig / popperConfig.

Do you have a use case on why we should trigger a re-render when changing the overlay config?